metrics/web/lib/main.dart

```diff
@@ -2,45 +2,33 @@ import 'dart:async';
-import 'package:metrics/common/presentation/injector/widget/injection_container.dart';
-import 'package:metrics/common/presentation/metrics_theme/config/color_config.dart';
-import 'package:metrics/common/presentation/metrics_theme/config/dimensions_config.dart';
-import 'package:metrics/common/presentation/metrics_theme/config/text_field_config.dart';
-import 'package:metrics/common/presentation/metrics_theme/config/text_style_config.dart';
-import 'package:metrics/common/presentation/metrics_theme/model/dark_metrics_theme_data.dart';
-import 'package:metrics/common/presentation/metrics_theme/model/light_metrics_theme_data.dart';
-import 'package:metrics/common/presentation/metrics_theme/widgets/metrics_theme_builder.dart';
-import 'package:metrics/common/presentation/navigation/metrics_route_information_parser.dart';
-import 'package:metrics/common/presentation/navigation/metrics_router_delegate.dart';
-import 'package:metrics/common/presentation/navigation/route_configuration/route_configuration_factory.dart';
-import 'package:metrics/common/presentation/navigation/state/navigation_notifier.dart';
-import 'package:metrics/common/presentation/routes/observers/firebase_analytics_route_observer.dart';
-import 'package:metrics/common/presentation/routes/observers/overlay_entry_route_observer.dart';
-import 'package:metrics/common/presentation/routes/observers/toast_route_observer.dart';
-import 'package:metrics/common/presentation/strings/common_strings.dart';
-import 'package:metrics/common/presentation/widgets/metrics_fps_monitor.dart';
-import 'package:metrics/common/presentation/widgets/metrics_scroll_behavior.dart';
+import 'package:metrics/common/presentation/metrics_app/metrics_app.dart';
 import 'package:metrics/metrics_logger/metrics_logger.dart';
 import 'package:metrics/metrics_logger/sentry/event_processors/user_agent_event_processor.dart';
 import 'package:metrics/metrics_logger/sentry/writers/sentry_writer.dart';
 import 'package:metrics/metrics_logger/writers/console_writer.dart';
 import 'package:metrics/metrics_logger/writers/logger_writer.dart';
 import 'package:metrics/util/favicon.dart';
-import 'package:provider/provider.dart';
 import 'package:universal_html/html.dart';
 
+import 'package:metrics/platform/stub/metrics_config/metrics_config_factory.dart'
+    if (dart.library.html) 'package:metrics/platform/web/metrics_config/metrics_config_factory.dart';
+
 Future<void> main() async {
   Favicon().setup();
 
   LoggerWriter writer;
 
+  final configFactory = MetricsConfigFactory();
+  final metricsConfig = configFactory.create();
+
   if (kReleaseMode) {
     final userAgent = window?.navigator?.userAgent;
     final eventProcessor = UserAgentEventProcessor(userAgent);
     writer = await SentryWriter.init(
-      const String.fromEnvironment('SENTRY_DSN'),
-      const String.fromEnvironment('SENTRY_RELEASE'),
+      metricsConfig?.sentryDsn,
+      metricsConfig?.sentryRelease,
+      metricsConfig?.sentryEnvironment,
```

Comment:
Could you please check cases when it could be `null` - I did some initial checks and usually it fails creating the object and throwing exception if something wrong in JS.